File: src/elasticsearch_hub_mcp/docs.py
"""Global documentation read/write system."""

from pathlib import Path

DOCS_FILE = Path(__file__).resolve().parent.parent.parent / "docs.md"


def get_docs() -> str:
    """Read the global docs file."""
    if not DOCS_FILE.exists():
        return "(No documentation yet. Use write_docs to add setup info.)"
    return DOCS_FILE.read_text()


def write_docs(content: str) -> str:
    """Write or append to the global docs file."""
    if DOCS_FILE.exists():
        existing = DOCS_FILE.read_text()
        DOCS_FILE.write_text(existing + "\n" + content)
    else:
        DOCS_FILE.write_text(content)
    return "Documentation updated."

File: src/docs.rs
use std::fs;
use std::path::Path;

/// Read the global docs file.
pub fn get_docs(docs_file: &Path) -> String {
    if !docs_file.exists() {
        return "(No documentation yet. Use write_docs to add setup info.)".to_string();
    }
    match fs::read_to_string(docs_file) {
        Ok(content) => content,
        Err(e) => format!("Error reading docs file: {}", e),
    }
}

/// Overwrite the global docs file.
pub fn write_docs(docs_file: &Path, content: &str) -> String {
    match fs::write(docs_file, content) {
        Ok(_) => "Documentation written.".to_string(),
        Err(e) => format!("Error writing docs file: {}", e),
    }
}

/// Append to the global docs file.
pub fn append_docs(docs_file: &Path, content: &str) -> String {
    let new_content = if docs_file.exists() {
        match fs::read_to_string(docs_file) {
            Ok(existing) => format!("{}\n{}", existing, content),
            Err(e) => return format!("Error reading docs file: {}", e),
        }
    } else {
        content.to_string()
    };

    match fs::write(docs_file, new_content) {
        Ok(_) => "Documentation appended.".to_string(),
        Err(e) => format!("Error writing docs file: {}", e),
    }
}
